xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-webjar/src/main/webjar/searchSuggest.js

```diff
@@ -131,10 +140,9 @@ var XWiki = (function (XWiki) {
             .insert(new Element('span', {'class':'suggestValue'}))
             .insert(new Element('span', {'class':'suggestInfo'}));
       this.noResultsMessage = new Element('div', {'class': 'hidden'})
-        .update("$escapetool.javascript($services.localization.render('core.widgets.suggest.noResults'))".escapeHTML());
-      var gotoSearchPageMessage = new Element('div')
-        .update("$escapetool.javascript($services.localization.render('core.widgets.suggest.showResults'))"
-        .escapeHTML());
+        .update(this._l10n['core.widgets.suggest.noResults']).escapeHTML();
```

Comment:
I don't see any ``escapeHTML`` function in http://api.prototypejs.org/dom/Element/index.html which is what [``update``](http://api.prototypejs.org/dom/Element/prototype/update/index.html) returns. Don't you want to escape the HTML before passing it to the ``update`` function?

Reply:
Sounds like @pjeanjean misread previous instructions before the changes: it used to be `"$escapetool.javascript($services.localization.render('core.widgets.suggest.noResults'))".escapeHTML()` and not `update("$escapetool.javascript($services.localization.render('core.widgets.suggest.noResults')").escapeHTML()`

So I guess the fix should be: `this._l10n['core.widgets.suggest.noResults'].escapeHTML()` if we do need escaping here.